### PR TITLE
Update XSLT README.md

### PR DESCRIPTION
**Pull request by Josar**
Added simple example with explanation.

Signed-off-by: josar <josuaarndt@live.de>

### CODE REVIEW

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 1]
```diff
@@ -2,21 +2,37 @@
 
 Transform input using the XML Stylesheet Language for Transformations (XSLT).
 
-It expects the transformation rule to be read from a file which is stored under the `transform` folder. 
+XSLT is a standard method to transform an XML structure into a document with the structure other structure.
+
+Trhe transformation expects the transformation rule to be read from a file which is stored under the `transform` folder.
```
Comment (Confectrician) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Typo -> `The`

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 2]
```diff
@@ -28,10 +44,60 @@ transform/helloworld.xsl:
...
+		<elevation>0</elevation>
+		<azimuth date="Fri, 18 Dec 2009 9:38 am PST" >450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+
+**transform/azimut.xsl**
+
+In the rule the tag `<xsl:stylesheet>` has to have the  `xmlns:xsl="http://www.w3.org/1999/XSL/Transform"` attribute and a second attribut `xmlns:` which has to be the same as the namespace for the input document. In the rule each step traversed along the path to the next tag has to prepend the `xmlns` namespace, here defined as `h`.
```
Comment (Confectrician) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Please break the text down to one line per Sentence.
We do it this way for better parsing and keeping git changes well documented.

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 3]
```diff
@@ -2,21 +2,33 @@
 
 Transform input using the XML Stylesheet Language for Transformations (XSLT).
 
-It expects the transformation rule to be read from a file which is stored under the `transform` folder. 
+XSLT is a standard method to transform an XML structures from one documentt into a new document with a other structure.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: A few grammar problems here.

Either "transform an XML structure" or "transform XML structures".
"document" (not "documentt")
"with another structure"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 4]
```diff
@@ -2,21 +2,33 @@
...
 To organize the various transformations one should use subfolders.
 
-## Example
+General transformation rule summary:
+
+* The directive `xsl:output` defines how the output document should be structured.
+* The directive `xsl:template` specifies matching attributes for the XML node to find. 
+* The `xsl:template` tag contains the rule which specifies what should be done.
+
+The Rule uses XPath to gather the XML node information for more information have a look at the XPath transformation.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Need a sentence break:
"to gather XML node information. For" 
or
"to gather XML node information; for"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 5]
```diff
@@ -2,21 +2,33 @@
...
+
+## Simple Example
 
-(from [here](https://en.wikipedia.org/wiki/Java_API_for_XML_Processing#Example))
+A simple but complete XLST transformation looks like follows, taken from [here](https://en.wikipedia.org/wiki/Java_API_for_XML_Processing#Example).
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "XSLT", not "XLST"
"looks like the following example, which was taken from ..."

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 6]
```diff
@@ -2,21 +2,33 @@
...
+**input XML**
 
 ```xml
 <?xml version='1.0' encoding='UTF-8'?>
 <root><node val='hello'/></root>
 ```
-
-transform/helloworld.xsl:
+**transform/helloworld.xsl**
+* `xsl:output`: transform incoming document into another XML-like document, without intendation.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "indentation"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 7]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
 
 ```xml
 <reRoot><reNode>hello world</reNode></reRoot>
 ```
 
-Other examples may be found [here](https://en.wikipedia.org/wiki/XSLT#XSLT_examples).
\ No newline at end of file
+## Advanced Example
+
+This example has an namespace defined, as you would find in real world applications, which has to be matched in the rule.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "has a namespace defined"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 8]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+
+**input XML**
+
+* The tag `<PTZStatus>` contains a namespace `xmlns=`
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: This should probably be reworded. `xmlns=` is not a namespace.

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 9]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+		<elevation>0</elevation>
+		<azimuth date="Fri, 18 Dec 2009 9:38 am PST" >450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+
+
+**transform/azimut.xsl**
+In the rule the tag `<xsl:stylesheet>` has to have the  `xmlns:xsl="http://www.w3.org/1999/XSL/Transform"` attribute and a second attribut `xmlns:` which has to be the same as the namespace for the input document.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "In the rule, ..."
"second attribute"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 10]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+In the rule each step traversed along the path to the next tag has to prepended with the `xmlns` namespace, here defined as `h`.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "In the rule, ..."
"has to be prepended"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 11]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+
+* `xsl:output` transform incoming document into another XML-like document, no intendation, **without XML**.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "indentation"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 12]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+* `xsl:template`: `match="/"` whole document.
+* Full path to node `azimuth` reading out `date` attribut.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "attribute"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 13]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+</xsl:stylesheet>
+```
+
+**Output Document**
+
+```
+Fri, 18 Dec 2009 9:38 am PST
+450
+```
+## Further Reading
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Need newline above header.

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 14]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+
+You can find a very good [introduction](https://www.w3schools.com/xml/xsl_intro.asp) and [tutorial](https://www.w3schools.com/xml/xsl_transformation.asp) at W3School.
+Extended introduction and more [examples](https://en.wikipedia.org/wiki/XSLT#XSLT_examples) at Wikipidia.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "Wikipedia"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 15]
```diff
@@ -28,10 +40,64 @@ transform/helloworld.xsl:
...
+A informative [tutorial](https://www.ibm.com/developerworks/library/x-xsltmistakes/) of common mistakes.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "An informative"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 16]
```diff
@@ -28,10 +41,66 @@ transform/helloworld.xsl:
...
 
-Other examples may be found [here](https://en.wikipedia.org/wiki/XSLT#XSLT_examples).
\ No newline at end of file
+## Advanced Example
+
+This example has a namespace defined, as you would find in real world applications, which has to be matched in the rule.
+
+**input XML**
+
+* The tag `<PTZStatus>` contains a attribute `xmlns=` which defines the namespace `http://www.hikvision.com/ver20/XMLSchema`.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "an attribute"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 17]
```diff
@@ -28,10 +41,66 @@ transform/helloworld.xsl:
...
+</PTZStatus>
+```
+
+
+**transform/azimut.xsl**
+In the rule, the tag `<xsl:stylesheet>` has to have an attribute `xmlns:xsl="http://www.w3.org/1999/XSL/Transform"` and a second attribute `xmlns:`. 
+This attribute has to be the same as the namespace for the input document.
+In the rule each step traversed along the path to the next tag has to be prepended with the `xmlns` namespace, here defined as `h`.
+
+* `xsl:output` transform incoming document into another XML-like document, no indendation, **without XML**.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "indentation"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 18]
```diff
@@ -2,21 +2,34 @@
 
 Transform input using the XML Stylesheet Language for Transformations (XSLT).
 
-It expects the transformation rule to be read from a file which is stored under the `transform` folder. 
+XSLT is a standard method to transform an XML structure from one document into a new document with an other structure.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "an other" should be "another". Or reword to say "a different"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 19]
```diff
@@ -2,21 +2,36 @@
...
+The Rule uses XPath to gather the XML node information.
+For more information have a look at the [XPath transformaton](https://docs.openhab.org/addons/transformations/xpath/readme.html) .
+
+## Examples
 
-(from [here](https://en.wikipedia.org/wiki/Java_API_for_XML_Processing#Example))
+### Basic Example
 
-input:
+A simple but complete XSLT transformaion looks like following example, which was taken from [here](https://en.wikipedia.org/wiki/Java_API_for_XML_Processing#Example).
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: "transformaion" should be "transformation"

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 20]
```diff
@@ -28,10 +43,66 @@ transform/helloworld.xsl:
...
+		<elevation>0</elevation>
+		<azimuth date="Fri, 18 Dec 2009 9:38 am PST" >450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+
+
+**transform/azimut.xsl**
+In the rule, the tag `<xsl:stylesheet>` has to have an attribute `xmlns:xsl="http://www.w3.org/1999/XSL/Transform"` and a second attribute `xmlns:`.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Needs a line break above this line

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 21]
```diff
@@ -28,10 +43,66 @@ transform/helloworld.xsl:
...
+**Output Document**
+
+```
+Fri, 18 Dec 2009 9:38 am PST
+450
+```
+
+## Further Reading
+
+You can find a very good [introduction](https://www.w3schools.com/xml/xsl_intro.asp) and [tutorial](https://www.w3schools.com/xml/xsl_transformation.asp) at W3School.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: Need to turn the 3 lines in this section into 3 bulleted items.

File: extensions/transform/org.eclipse.smarthome.transform.xslt/README.md [hunk 22]
```diff
@@ -28,10 +45,66 @@ transform/helloworld.xsl:
...
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="2.0" 
+xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+xmlns:h="http://www.hikvision.com/ver20/XMLSchema">
+   <xsl:output method="xml" indent="no" encoding="UTF-8" omit-xml-declaration="yes"  />
+   <xsl:template match="/">
+      <xsl:value-of select="/h:PTZStatus/h:AbsoluteHigh/h:azimuth/@date" />
+      <xsl:text>&#10;</xsl:text>
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/README.md]: I think you also should explain this line in the text above.

**Pull request merged**